lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -0,0 +1,146 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Cardano.Wallet.DB.Sqlite where
+
+import Prelude
+
+import Cardano.Wallet.DB.SqliteTypes
+    ( AddressScheme, TxId )
+import Data.Text
+    ( Text )
+import Data.Time.Clock
+    ( UTCTime )
+import Data.Word
+    ( Word32 )
+import Database.Persist.TH
+import GHC.Generics
+    ( Generic (..) )
+
+import qualified Cardano.Wallet.Primitive.Types as W
+
+-- fixme: need tables for wallet AddressPool
+
+share
+    [ mkPersist sqlSettings { mpsPrefixFields = False }
+    , mkMigrate "migrateAll"
+    ]
+    [persistLowerCase|
+
+-- Wallet IDs, address discovery state, and metadata.
+Wallet
+    walTableId                 W.WalletId     sql=wallet_id
+    walTableName               Text           sql=name
+    walTablePassLastUpdatedAt  UTCTime        sql=passphrase_last_updated_at
+    walTableStatus             W.WalletState  sql=status
+    walTableDelegation         Text Maybe     sql=delegation
+    walTableAddressScheme      AddressScheme  sql=address_discovery
+
+    Primary walTableId
+    deriving Show Generic
+
```

Comment:
I have a question about the naming here.

Is there a good reason to use the word `Table` in every field name?

Why not just make sure that each field name starts with the name of the table as a prefix, but in lower case.

So:
```hs
Foo
    fooX
    fooY
```

For example:

```hs
Wallet
    walletId                 W.WalletId     sql=wallet_id
    walletName               Text           sql=name
    walletPassLastUpdatedAt  UTCTime        sql=passphrase_last_updated_at
    walletStatus             W.WalletState  sql=status
    walletDelegation         Text Maybe     sql=delegation
    walletAddressScheme      AddressScheme  sql=address_discovery
```
```hs
TxInput
    txInputTxId         TxId        sql=tx_id
    txInputWalletId     W.WalletId  sql=wallet_id
    txInputSourceTxId   TxId        sql=source_id
    txInputSourceIndex  Word32      sql=source_index
```

Reply:
I like putting `Table` especially in the context of writing encoders/decoders. Then, it is immediately visible to whether field belongs to the generated type from template haskell or to other modules. That's the rationale behind introducing it. Nevertheless, I am not about to fight for this ;-)